File: app/src/main/java/com/permissionnanny/dagger/ActivityComponent.java
package com.permissionnanny.dagger;

import com.permissionnanny.ConfirmRequestActivity;
import com.permissionnanny.missioncontrol.AppControlActivity;
import dagger.Component;

/**
 *
 */
@ContextComponent.ContextScope
@Component(modules = {ContextModule.class, ActivityModule.class}, dependencies = {AppComponent.class})
public interface ActivityComponent {

    void inject(AppControlActivity victim);

    void inject(ConfirmRequestActivity victim);
}

File: app/src/main/kotlin/com/permissionnanny/dagger/ActivityComponent.kt
package com.permissionnanny.dagger

import com.permissionnanny.ConfirmRequestActivity
import com.permissionnanny.missioncontrol.AppControlActivity
import dagger.Component

/**

 */
@ContextComponent.ContextScope
@Component(modules = arrayOf(ContextModule::class, ActivityModule::class), dependencies = arrayOf(AppComponent::class))
interface ActivityComponent {

    fun inject(victim: AppControlActivity)

    fun inject(victim: ConfirmRequestActivity)
}
